src/ROB.java

```diff
@@ -0,0 +1,119 @@
+import java.util.ArrayList;
+
+public class ROB
+{
+	private final int ROB_SIZE;
+	private int cur_size;
+	// The queue is inclusive between front_i and back_i, under mod ROB_SIZE
+	private int front_i;
+	private int back_i;
+
+	private ArrayList<Instruction> queue = new ArrayList<Instruction>();
+
+	public ROB(final int size)
+	{
+		ROB_SIZE = size;
+		cur_size = 0
+		front_i = 0;
+		back_i = ROB_SIZE - 1;
+	}
+
+	public boolean enqueue(Instruction inst)
```

Comment:
I added `ROB::killInstructionsBetween()` to handle that. So after the BranchUnit decides that instructions need to be killed (in the Simulator), then it can call `ROB::killInstructionsBetween()`, with `addr1 = pc+4` and `addr2 = target-1`, or `addr1 = target` and `addr2 = -1`.